### ISSUE TITLE
Snap-based DogeHouse Electron app for Linux users

### ISSUE THREAD

**Issue by ajhalili2006** [benawad/dogehouse#575]
## To-do

* [x] Create new repo/fork based on the[ `github:snapcrafters/fork-and-rename-me`](https://github.com/snapcrafters/fork-and-rename-me) template.
* [x] Port Electron app to Snap as part of build
* [x] Edit the `snapcraft.yaml` file to adjust to the app's needs (plugs, slots, etc.)
* [x] Update README
* [ ] Complete remaining tasks in https://github.com/snapcrafters/fork-and-rename-me/blob/master/README.md#remaining-tasks (optionally transfer ownership to `snapcrafters` unless Ben say no)

## Additional Resources

**Related issue**: #172

**Creating an Electron Snap app**: https://snapcraft.io/docs/build-snaps/electron

**Comment by ajhalili2006** [comment]
I successfully created the repo from the template and it's on https://github.com/MadeByThePinsHub/dogehouse-electron-app-snap.

Once the community is done porting stuff to Snaps, I'll handle the maintainership to @benawad.

**Comment by overlisted** [comment]
please make it for flatpak

**Comment by yummyweb** [comment]
What is the status for this issue? Please update (if any). Currently, the debian version built from electron-builder gave some errors so I am looking into that.

**Comment by ajhalili2006** [comment]
> What is the status for this issue? Please update (if any). Currently, the debian version built from electron-builder gave some errors so I am looking into that.

Hey, @yummyweb. Currently, the build using `snapcraft.yaml`'s `parts.dogehouse.override-build` turned into chaos, so maybe I'll comment out the `override-build` and rely on `electron-builder --linux snap` option.

**Comment by yummyweb** [comment]
Hey @AndreiJirohHaliliDev2006 there is a problem with the debian version built by `electron-builder` so I advice don't rely on `electron-builder` and if you face a problem similar to issue, #1112 then please let me know and comment on the issue.

**Comment by ajhalili2006** [comment]
> Hey @AndreiJirohHaliliDev2006 there is a problem with the debian version built by `electron-builder` so I advice don't rely on `electron-builder` and if you face a problem similar to issue, #1112 then please let me know and comment on the issue.

@yummyweb So for now, do I need to use the deb file from the Releases tab and repackage into Snap for now?

**Comment by amitojsingh366** [comment]
@yummyweb @AndreiJirohHaliliDev2006 there isn't anything wrong with `electron-builder` i believe it's to do with our dependency `iohook` *(i'm pretty sure)*

**Comment by yummyweb** [comment]
Yes yes iohook is the problem and more specifically with the xkb dependency of iohook

**Comment by amitojsingh366** [comment]
I think this is pointless *(i might be wrong tho)* have a look at this https://www.electronjs.org/docs/tutorial/snapcraft

**Comment by ajhalili2006** [comment]
> I think this is pointless _(i might be wrong tho)_ have a look at this https://www.electronjs.org/docs/tutorial/snapcraft

Hi, @amitojsingh366! Maybe you're wrong tho, but I'll either merge back the `snap` directory into `baklava/snap` here soon or transfer the repo to Ben.

For now, I have other stuff I may need to do, so my apologies for the delay to reply and also updating the snap.

**Comment by amitojsingh366** [comment]
Umm i meant to say doesn't electron have inbuilt support for snaps
https://www.electronjs.org/docs/tutorial/snapcraft

**Comment by ajhalili2006** [comment]
> Umm i meant to say doesn't electron have inbuilt support for snaps
> https://www.electronjs.org/docs/tutorial/snapcraft

@amitojsingh366 Well there's support, but will merge into this repo soon.

**Comment by amitojsingh366** [comment]
@AndreiJirohHaliliDev2006 dogehouse now supports snaps. https://snapcraft.io/dogehouse

**Comment by amitojsingh366** [comment]
ill be closing this issue now since a snap has been published.
